generative_data_prep/processors/metrics.py

```diff
@@ -0,0 +1,129 @@
+        return total_articles_dropped / self.articles
+
+    @property
+    def percent_articles_dropped_from_prompt(self) -> float:
+        """The percent of the articles dropped due to having only prompt tokens (no completion)."""
+        return self.articles_dropped_from_all_prompt / self.articles
+
+    @property
+    def percent_articles_dropped_from_packing(self) -> float:
+        """The percent of the articles that are dropped due to packing style."""
+        return self.articles_dropped_from_all_prompt / self.articles
+
+    @property
+    def averge_prompt_length(self) -> float:
+        """The average number of tokens per prompt."""
+        return self.prompt_tokens / self.articles
+
+    @property
+    def average_completion_length(self) -> float:
+        """The average number of tokens per completion."""
+        return self.completion_tokens / self.articles
+
+    @property
+    def sequence_utilization(self) -> float:
+        """What percent of the tokens are not padding."""
+        return 1 - (self.padding_tokens / self.tokens)
+
+    @property
+    def sequence_completion_utilization(self) -> float:
+        """What percent of the tokens are completions."""
+        return self.completion_tokens / self.tokens
+
+    def _to_str_percent(self, value: int) -> str:
+        percent = round(value * 100, 2)
+        return f"{percent:.2f}%"
+
+    def __str__(self):
+        """String representation of metrics."""
+        table = [
+            ["Sequences", self.sequences],
+            ["Articles", self.articles],
+            ["Tokens", self.tokens],
+            ["Prompt Tokens", self.prompt_tokens],
+            ["Completion Tokens", self.completion_tokens],
+            ["Padding Tokens", self.padding_tokens],
+            ["Average Completion Length", round(self.average_completion_length, 2)],
+            ["Average Prompt Length", round(self.averge_prompt_length, 2)],
+            ["Prompt Completion Pairs", self.prompt_completion_pairs],
+            ["Tokens Dropped", self.tokens_dropped],
+            ["Articles Dropped From Packing", self.articles_dropped_from_packing],
+            ["Percent Dropped From Packing", self._to_str_percent(self.percent_articles_dropped_from_packing)],
+            ["Articles Dropped From All Prompt", self.articles_dropped_from_all_prompt],
+            ["Percent Dropped From All Prompt", self._to_str_percent(self.percent_articles_dropped_from_prompt)],
+            ["Sequence Utilization", self._to_str_percent(self.sequence_utilization)],
+            ["Sequence Completion Utilization", self._to_str_percent(self.sequence_completion_utilization)]
+        ]
+        ret = "=================================================="
+        for name, value in table:
+            ret += "\n" + f"{name}: {value}"
+        ret += "\n" + "=================================================="
+        return ret
+
+        # return tabulate(table, "fancy_outline")
```

Comment:
I want to replace this with tabulate to get a nice table print, but I am not sure how to get the dependency working. I asked @snova-ranjanl to help